x/evm/ante/fee.go

```diff
@@ -55,13 +57,29 @@ func (fc EVMFeeCheckDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 		}
 	}
 
-	// fee + value
-	anteCharge := txData.Cost() // this would include blob fee if it's a blob tx
-
-	senderEVMAddr := evmtypes.MustGetEVMTransactionMessage(tx).Derived.SenderEVMAddr
 	// check if the sender has enough balance to cover fees
-	if state.NewDBImpl(ctx, fc.evmKeeper, true).GetBalance(senderEVMAddr).Cmp(anteCharge) < 0 {
-		return ctx, sdkerrors.ErrInsufficientFunds
+	etx, _ := msg.AsTransaction()
+	emsg := fc.evmKeeper.GetEVMMessage(ctx, etx, msg.Derived.SenderEVMAddr)
+	stateDB := state.NewDBImpl(ctx, fc.evmKeeper, false)
+	gp := fc.evmKeeper.GetGasPool()
+	blockCtx, err := fc.evmKeeper.GetVMBlockContext(ctx, gp)
+	if err != nil {
+		return ctx, err
+	}
+	cfg := evmtypes.DefaultChainConfig().EthereumConfig(fc.evmKeeper.ChainID(ctx))
+	txCtx := core.NewEVMTxContext(emsg)
+	evmInstance := vm.NewEVM(*blockCtx, txCtx, stateDB, cfg, vm.Config{})
+	stateDB.SetEVM(evmInstance)
+	st := core.NewStateTransition(evmInstance, emsg, &gp, true)
+	if err := st.BuyGas(); err != nil {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrInsufficientFunds, err.Error())
```

Comment:
are there any other errors possible here (always insufficient funds)?

Reply:
yea insufficient funds is the only possible error https://github.com/sei-protocol/go-ethereum/blob/master/core/state_transition.go#L259